strings-csharp/ReplaceSpecialCharactersInString/ReplaceSpecialCharactersInString/StringReplacer.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+
+
+namespace ReplaceSpecialCharactersInString
+{
+    public class StringReplacer
+    {
+        public static string ReplaceUsingStringReplace(string originalString,
+            char oldChar, char newChar)
+        {
+            return originalString.Replace(oldChar, newChar);
+        }
+
+
+        public static string ReplaceUsingStringBuilder(string originalString,
+            char oldChar, char newChar)
+        {
+            var sb = new StringBuilder(originalString);
+
+            sb.Replace(oldChar, newChar);
+
+            return sb.ToString();
+        }
+
+
+        public static string ReplaceUsingRegex(string originalString,
+            string oldChar, string newChar)
+        {
+            var pattern = $"[{Regex.Escape(oldChar)}]";
+
+            return Regex.Replace(originalString, pattern, newChar);
+        }
+
+
+        public static string ReplaceUsingSpan(string originalString,
+            char oldChar, char newChar)
+        {
+            var originalChars = originalString.ToCharArray();
+
+            var span = new Span<char>(originalChars);
+
+            span.Replace(oldChar, newChar);
+
+            return new string(span.ToArray());
+        }
+
+
+        public static string ReplaceUsingInefficientMultipleReplacementsStringReplace(
+            string originalString, char oldChar, char newChar)
+        {
+            var replacedString = originalString
+                .Replace(newChar, oldChar)
+                .Replace("c", "")
+                .Replace(oldChar, newChar);
+
+            return replacedString;
+        }
+
+
+        public static string ReplaceUsingMemoryImpactStringReplace(
+               string originalString, char oldChar, char newChar)
+        {
+            var iterations = 10;
+            var replacedString = originalString;
+
+            for (int i = 0; i < iterations; i++)
```

Comment:
This is again not correct. This type of code should not be present in library. This implementation must be in the caller.